lib/kubeclient/common.rb

```diff
@@ -416,6 +428,18 @@ def api
       JSON.parse(response)
     end
 
+    def self.restclient_read_timeout_option
```

Comment:
> What happens if we pass in both options?
￼ 
Interesting, you mean:
```
        open_timeout: @timeouts[:open],
        read_timeout: @timeouts[:read], # rest-client 2.x
        timeout: @timeouts[:read]       # rest-client 1.x
```
Yes, that works AFAICT!  I've attempted something similar but not exactly this.

However, while our code would look simpler what actually happens will be harder to reason about.
It'll also be harder to test, due to https://github.com/rest-client/rest-client/issues/591 I discovered while trying.  I even tried testing it upstream in rest-client itself but gave up (https://github.com/rest-client/rest-client/pull/595).

=> Bottom line: I'd prefer the method probing.